src/scripts/main.js

```diff
@@ -1,3 +1,102 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('.field');
+const rowRemove = document.querySelector('.remove-row');
+const rowAdd = document.querySelector('.append-row');
+const columnAdd = document.querySelector('.append-column');
+const columnRemove = document.querySelector('.remove-column');
+
+document.addEventListener('click', (e) => {
+  const target = e.target.closest('.button');
+
+  if (!target) {
+    return;
+  }
+
+  changeTable(target);
+});
+
+function changeTable(elem) {
+  const addRow = elem.closest('.append-row');
+  const addColumn = elem.closest('.append-column');
+  const removeRow = elem.closest('.remove-row');
+  const removeColumn = elem.closest('.remove-column');
+
+  switch (elem) {
```

Comment:
The switch statement is using 'elem' as the expression, but 'elem' is an element, not a string or number. This will not work as expected because 'case' clauses are comparing objects by reference, not by their properties or classes. Consider using if-else statements to check if 'elem' matches 'addRow', 'removeRow', etc.